Change the default root project location to be consistent with build tools

j-em opened the pull request:
Hope this helps with https://github.com/jasongilman/proto-repl/issues/47
This change the default root directory of a project to a location where a `project.clj` file can be found - defaulting to null when it can't be found.
This is should be similar to how `clojure-mode` (and therefore `cider`) works in emacs.

Comments?

j-em committed:
add method to get the real root of a project relative to lein project file

jasongilman commented:
Thanks for this pull request! I do have a few comments that I think should be resolved first before this gets merged. I'll add them shortly.

j-em commented:
Thanks for the comments ! Okay so I've removed the matches for boot files and I added a new `limit` parameter.
Then I added a "safeguard" at the beginning of the function that checks if the limit is < 100, and if the parent directory of the current path and the current path itself are _not_ the same. Provided that the implementation of the `path` module is consistent across all the supported platform (which it should be), I think this should fix the issues related to finding the root of the fs without hard-coding anything manually.

What do you think?

jasongilman commented:
This looks good to me. I'll merge it into master and test it.

jasongilman commented:
Thanks again!
